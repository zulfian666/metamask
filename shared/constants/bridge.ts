import { CHAIN_IDS } from './network';

// TODO read from feature flags
export const ALLOWED_BRIDGE_CHAIN_IDS = [
  CHAIN_IDS.MAINNET,
  CHAIN_IDS.BSC,
  CHAIN_IDS.POLYGON,
  CHAIN_IDS.ZKSYNC_ERA,
  CHAIN_IDS.AVALANCHE,
  CHAIN_IDS.OPTIMISM,
  CHAIN_IDS.ARBITRUM,
  CHAIN_IDS.LINEA_MAINNET,
  CHAIN_IDS.BASE,
];

export const BRIDGE_DEV_API_BASE_URL = 'https://bridge.dev-api.cx.metamask.io';
export const BRIDGE_PROD_API_BASE_URL = 'https://bridge.api.cx.metamask.io';
export const BRIDGE_API_BASE_URL = process.env.BRIDGE_USE_DEV_APIS
  ? BRIDGE_DEV_API_BASE_URL
  : BRIDGE_PROD_API_BASE_URL;

export const BRIDGE_CLIENT_ID = 'extension';
