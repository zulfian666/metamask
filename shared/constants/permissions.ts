export const CaveatTypes = Object.freeze({
  restrictReturnedAccounts: 'restrictReturnedAccounts' as const,
  restrictNetworkSwitching: 'restrictNetworkSwitching' as const,
});

export const RestrictedEthMethods = Object.freeze({
  eth_accounts: 'eth_accounts',
});

export const RestrictedMethods = Object.freeze({
  eth_accounts: 'eth_accounts',
  snap_dialog: 'snap_dialog',
  snap_notify: 'snap_notify',
  snap_manageState: 'snap_manageState',
  snap_getBip32PublicKey: 'snap_getBip32PublicKey',
  snap_getBip32Entropy: 'snap_getBip32Entropy',
  snap_getBip44Entropy: 'snap_getBip44Entropy',
  snap_getEntropy: 'snap_getEntropy',
  snap_getLocale: 'snap_getLocale',
  wallet_snap: 'wallet_snap',
  ///: BEGIN:ONLY_INCLUDE_IF(keyring-snaps)
  snap_manageAccounts: 'snap_manageAccounts',
  ///: END:ONLY_INCLUDE_IF
} as const);

// ConnectionPermission is pseudo permission used to make possible
// displaying pre-approved connections in the UI seamlessly, alongside other permissions.
export const ConnectionPermission = Object.freeze({
  connection_permission: 'connection_permission',
});

export * from './snaps/permissions';
