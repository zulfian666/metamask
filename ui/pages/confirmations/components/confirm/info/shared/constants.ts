export const HEX_ZERO = '0x0';
