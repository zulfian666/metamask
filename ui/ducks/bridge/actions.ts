import { swapsSlice } from '../swaps/swaps';
import { bridgeSlice } from './bridge';

// Bridge actions

// eslint-disable-next-line no-empty-pattern
const {} = swapsSlice.actions;

export const { setToChain } = bridgeSlice.actions;
