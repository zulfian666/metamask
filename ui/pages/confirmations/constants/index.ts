import { CHAINLIST_CHAIN_IDS_MAP } from '../../../../shared/constants/network';

export const EIP712_PRIMARY_TYPE_PERMIT = 'Permit';

export const IGNORE_GAS_LIMIT_CHAIN_IDS = [CHAINLIST_CHAIN_IDS_MAP.MANTLE];

export const TYPED_SIGNATURE_VERSIONS = {
  V1: 'V1',
  V3: 'V3',
  V4: 'V4',
};
