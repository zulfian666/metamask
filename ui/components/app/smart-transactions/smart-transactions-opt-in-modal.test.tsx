import React from 'react';
import { fireEvent } from '@testing-library/react';
import thunk from 'redux-thunk';
import configureMockStore from 'redux-mock-store';
import { useHistory } from 'react-router-dom';

import {
  renderWithProvider,
  createSwapsMockStore,
} from '../../../../test/jest';
import { setSmartTransactionsOptInStatus } from '../../../store/actions';
import { ADVANCED_ROUTE } from '../../../helpers/constants/routes';
import SmartTransactionsOptInModal from './smart-transactions-opt-in-modal';

const middleware = [thunk];

jest.mock('../../../store/actions');

jest.mock('react-router-dom', () => ({
  ...jest.requireActual('react-router-dom'),
  useHistory: jest.fn(() => []),
}));

describe('SmartTransactionsOptInModal', () => {
  it('displays the correct text in the modal', () => {
    const store = configureMockStore(middleware)(createSwapsMockStore());
    const { getByText, container } = renderWithProvider(
      <SmartTransactionsOptInModal
        isOpen={true}
        hideWhatsNewPopup={jest.fn()}
      />,
      store,
    );
    expect(getByText('Enable')).toBeInTheDocument();
    expect(getByText('Manage in settings')).toBeInTheDocument();
    expect(container).toMatchSnapshot();
  });

  it('calls setSmartTransactionsOptInStatus with false when the "Manage in settings" link is clicked and redirects to Advanced Settings', () => {
    (setSmartTransactionsOptInStatus as jest.Mock).mockImplementationOnce(() =>
      jest.fn(),
    );
    const historyPushMock = jest.fn();
    (useHistory as jest.Mock).mockImplementationOnce(() => ({
      push: historyPushMock,
    }));
    const store = configureMockStore(middleware)(createSwapsMockStore());
    const { getByText } = renderWithProvider(
      <SmartTransactionsOptInModal
        isOpen={true}
        hideWhatsNewPopup={jest.fn()}
      />,
      store,
    );
    const manageInSettingsLink = getByText('Manage in settings');
    fireEvent.click(manageInSettingsLink);
    expect(setSmartTransactionsOptInStatus).toHaveBeenCalledWith(false);
    expect(historyPushMock).toHaveBeenCalledWith(
      `${ADVANCED_ROUTE}#smart-transactions`,
    );
  });

  it('calls setSmartTransactionsOptInStatus with true when the "Enable" button is clicked', () => {
    (setSmartTransactionsOptInStatus as jest.Mock).mockImplementationOnce(() =>
      jest.fn(),
    );
    const store = configureMockStore(middleware)(createSwapsMockStore());
    const { getByText } = renderWithProvider(
      <SmartTransactionsOptInModal
        isOpen={true}
        hideWhatsNewPopup={jest.fn()}
      />,
      store,
    );
    const enableButton = getByText('Enable');
    fireEvent.click(enableButton);
    expect(setSmartTransactionsOptInStatus).toHaveBeenCalledWith(true);
  });
});
