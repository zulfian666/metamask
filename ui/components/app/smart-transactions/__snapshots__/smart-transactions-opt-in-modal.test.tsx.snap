// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SmartTransactionsOptInModal displays the correct text in the modal 1`] = `<div />`;
